### PR TITLE
Adding plotting options to the Muon Analysis GUI

### LINKED ISSUE #26885: muon plotting options
And a plotting options to muon GUIs (have it where the collapsible menus will be).

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
This PR represents the final plotting improvement in the Muon Analysis GUI through the addition of plotting options. These options allow the axis bounds, and errors to be changed easily through the addition of an options widget. This PR is based on the work undertaken in #27529 and therefore should not be merged before #27529 is. 

**To test**
```
Example workflow
1. Open the Muon Analysis 2 GUI
2. Load in some data (e.g MUSR - 62260)
3. Change the x-axis bounds, autoscale the data
4. Change the y-axis bounds
5. Do a tiled plot
6. Change the chosen subplot in the options - Change axis limits, autoscale
7. Zoom using the figure toolbar, and verify the axis updates and can be autoscaled.
8. Plot and un-plot errors
```

Fixes #26885 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/Muon/workspace_finder_test.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        self.workspace_finder.get_time_workspaces_to_plot = mock.Mock(return_value=["62260;fwd"])
+
+        self.workspace_finder.get_workspaces_to_plot(True, "Asymmetry")
+
+        self.assertEquals(self.workspace_finder.get_freq_workspaces_to_plot.call_count, 0)
+        self.assertEquals(self.workspace_finder.get_time_workspaces_to_plot.call_count, 1)
+
+    def test_get_workspace_to_plot_time(self):
+        runs = [[62260]]
+        self.context.data_context.current_runs = runs
```
Comment (Harrietbrown) [inline scripts/test/Muon/workspace_finder_test.py]: You can set current runs directly, no need to use a runs variable.

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
If you change it to tile plot, then resize the axes indevidualy toggeling the error bar will automatically resize all windows to the combined plot (the plot before making it tiled.)

File: scripts/test/Muon/plotting_widget_freq_test.py [hunk 2]
```diff
@@ -45,9 +46,9 @@ def test_time_plot_in_FDA(self):
...
 
-        self.model.add_workspace_to_plot.assert_any_call(self.presenter.get_workspace_plot_axis(),
-                                                         self.workspace_list[1], [0], errors=True,
-                                                         plot_kwargs=plot_kwargs)
+        self.model.add_workspace_to_plot.assert_called_with(self.presenter.get_workspace_plot_axis(),
+                                                            self.workspace_list[1], [0], errors=True,
+                                                            plot_kwargs=plot_kwargs)
 
     def test_plot_type_changed(self):
         self.view.get_selected.return_value = "Asymmetry"
```
[the remark this answers is not in the document]
Comment (StephenSmith25) [reply scripts/test/Muon/plotting_widget_freq_test.py]: Good spot, it was a pointless assertion.

File: scripts/Muon/GUI/Common/plotting_widget/plotting_widget_model.py [hunk 3]
```diff
@@ -190,25 +189,60 @@ def replace_workspace_plot(self, workspace_name, axis):
...
-            ymin, ymax = self._get_autoscale_y_limits(axes, 0, 15)
-            plt.setp(axes, xlim=[0, 15.0], ylim=[ymin, ymax])
-        if domain == "Frequency":
-            ymin, ymax = self._get_autoscale_y_limits(axes, 0, 50)
-            plt.setp(axes, xlim=[0, 50.0], ylim=[ymin, ymax])
+    def _remove_all_data_workspaces_from_plot(self, axes):
+        workspaces_to_remove = self.plotted_workspaces
+        for workspace in workspaces_to_remove:
+            self.remove_workspace_from_plot(workspace, axes)
+        workspaces_to_remove = self.plotted_fit_workspaces
```
[the remark this answers is not in the document]
Comment (StephenSmith25) [reply scripts/Muon/GUI/Common/plotting_widget/plotting_widget_model.py]: Does this refer to the remove_workspaces from plot bit? The second call updates the workspaces_to_remove variable, and then removes the fit workspace.

File: scripts/test/Muon/workspace_finder_test.py (hunk 1 above)
Comment (StephenSmith25) [reply scripts/test/Muon/workspace_finder_test.py]: I was favoring a variable, so I didn't have to rewrite the same information [[62260]] in the assert calls.

File: scripts/Muon/GUI/Common/plotting_widget/plotting_widget_model.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (StephenSmith25) [reply scripts/Muon/GUI/Common/plotting_widget/plotting_widget_model.py]: The second call is related to removing the fitted workspaces, as I reassign workspaces_to_remove above the second loop.

[the remark this answers is not in the document]
Comment (Harrietbrown) [reply scripts/Muon/GUI/Common/plotting_widget/plotting_widget_model.py]: oops I missread that.

**Comment by StephenSmith25** [comment]
> If you change it to tile plot, then resize the axes indevidualy toggeling the error bar will automatically resize all windows to the combined plot (the plot before making it tiled.)

I've now prevented the error box from resizing everything.

**Review by Harrietbrown — DISMISSED** [review]
The plot gui looks good, the error bars appear as expected and changing the axis limits works. Good job.

**Comment by gvardany** [comment]
PR #27529 is merged, can you merge master on this, just to make sure there are no conflicts?

**Comment by StephenSmith25** [comment]
@Harrietbrown Could you re-approve this as I had to merge in master.

**Review by Harrietbrown — APPROVED** [review]
It still works after merging maste rinto it.

**Pull request merged**